### DISCUSSION TITLE
Site permission for editing users

### DISCUSSION THREAD

**Discussion by janhenckens** [craftcms/cms#8725] [Ideas]
### Description
Working on a multisite where each users only has permissions to their own site and they also have permission to add new users. We created a group with permissions for each site and only allow them to add users to their own site.

When just viewing users however, they can see **and** edit users from other sites and we don't really want that 🙂

Users or usergroups aren't defined on a per-site basis but I wonder what the best solution for this would be. 

A work-around could be to use `Element::EVENT_REGISTER_SOURCES` on the `User` element to only show users who have permission for the site(s) that the current user has permissions to

Curious what your thoughts are on this.
Cheers
Jan

### Additional info

- Craft version: 3.2.10
- PHP version: 7.3

**Comment by brandonkelly** [comment]
Hm interesting one. To enforce this we would basically need to add new “Edit Users” permissions for each individual site.  I wonder if that would be annoying or desired for other installs.

**Comment by gtettelaar** [comment]
@brandonkelly I'm assuming you are talking about a `Edit users for user group X` permission? Users are not stored on a site-by-site basis so how would a site based permission work?

**Comment by brandonkelly** [comment]
@gtettelaar more like `Edit users that are permitted to edit Site X` – but some issue (users can be permitted to edit content on multiple sites). It’s not pretty though.

**Comment by janhenckens** [comment]
Making it `Edit users for user group` would keep it in line with `Assign users to user group` and  although that would accomplish what we need in this use-case/project, that only works when user groups match up with sites, which won't always be the case.

Regardless of the user's site permissions, I've always found it strange that a user's ability to add new users can be limited to a group but the editing of users can not. Maybe making adding and editing 1 permission would be good here. But that would be big change versus how it works now so not ideal either.

**Comment by gtettelaar** [comment]
> Regardless of the user's site permissions, I've always found it strange that a user's ability to add new users can be limited to a group but the editing of users can not.

I'm not sure about that. Assigning users to group's is fundamentally about _granting_ a user permissions. The reason that there is a `Assign User to User Group X` is that there needs to be some mechanism to prevent permission escalation. Editing a user was always related to the general `Edit users` permission - because User content is _general_ (i.e. not related to sites) I always thought this made sense. Personal preference I suppose ¯\_(ツ)_/¯. 

> although that would accomplish what we need in this use-case/project, that only works when user groups match up with sites, which won't always be the case.

Arguably that needs to be the responsibility of the developer working on the Site IMO. 

> @gtettelaar more like Edit users that are permitted to edit Site X – but some issue (users can be permitted to edit content on multiple sites). It’s not pretty though.

Doing this seems like it would result in adding something pretty significant to the core to only solve a pretty specific problem/use-case, especially with certain changes that are on the way (see below). One of the reasons _I_ think Craft is awesome is because it doesn't get _too_ opinionated or specific about certain things - [and it's pretty strict about keeping it that way](https://github.com/craftcms/cms/blob/develop/CONTRIBUTING.md#core-enhancements). Unless there is a big demand for this specific permission, I would say it should go into a module/plugin. 

<hr>

Personally I would say this issue needs to be addressed in conjunction with #4463 or something similar. Once Users become multi-site enabled it should be possible to restrict editing of User content to users with the `editSite:{siteUid}` permission. If we allow Users to be disabled for a specific site we _should_ also be able to keep them out of the `/admin/users` index screen. Which then should address the FR, I think...
